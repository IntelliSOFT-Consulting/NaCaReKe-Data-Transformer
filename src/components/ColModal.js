/* eslint-disable react/forbid-prop-types */
import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'antd';
import ColumnForm from './ColumnForm';

export default function ColModal({
  handleChange,
  onFinish,
  visible,
  cols,
  setVisible,
  form,
}) {
  const formRef = useRef(null);
  return (
    <Modal
      title="Edit data"
      visible={visible}
      onOk={() => formRef.current.submit()}
      onCancel={() => setVisible(false)}
      width={1000}
      okText="Submit"
    >
      <ColumnForm
        formRef={formRef}
        onFinish={onFinish}
        handleChange={handleChange}
        cols={cols}
        form={form}
      />
    </Modal>
  );
}

ColModal.propTypes = {
  handleChange: PropTypes.func.isRequired,
  onFinish: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
  cols: PropTypes.array.isRequired,
  setVisible: PropTypes.func.isRequired,
  form: PropTypes.object.isRequired,
};
