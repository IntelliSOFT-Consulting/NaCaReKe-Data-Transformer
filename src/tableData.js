/* eslint-disable import/no-mutable-exports */
let data = [
  [
    'RECS',
    'RECS (desc)',
    'CHEC',
    'CHEC (desc)',
    'AGE',
    'ADDR',
    'ADDR (cat)',
    'ADDR (desc)',
    'INCID',
    'TOP',
    'TOP (cat)',
    'TOP (desc)',
    'LTERALITY',
    'LTERALITY (desc)',
    'MOR',
    'MOR (desc)',
    'BEH',
    'BEH (desc)',
    'BAS',
    'BAS (desc)',
    'MPCODE',
    'MPSEQ',
    'MPTOT',
    'I10',
    'UPDATE',
    'ICCC',
    'OBSOLETEFLAGTUMOURTABLE',
    'TUMOURID',
    'PATIENTIDTUMOURTABLE',
    'PATIENTRECORDIDTUMOURTABLE',
    'TUMOURUPDATEDBY',
    'TUMOURUNDUPLICATIONSTATUS',
    'LABNUMBER',
    'LABREPORTAVAILABLEPTFILE',
    'LABREPORTAVAILABLEPTFILE (desc)',
    'CD4COUNT',
    'STATUSINDICATEDONCLINICALNOTES',
    'STATUSINDICATEDONCLINICALNOTES (desc)',
    'SPECIFICALLYNEGATIVE',
    'SPECIFICALLYNEGATIVE (desc)',
    'SPECIFICALLYPOSITIVE',
    'SPECIFICALLYPOSITIVE (desc)',
    'OTHERCONCURRENTILLNESS',
    'OTHERADDRESS',
    'GRADE',
    'GRADE (desc)',
    'STAGE',
    'STAGE (desc)',
    'REGNO',
    'PERS',
    'PERS (desc)',
    'FIRSTN',
    'MIDDLNAME',
    'FAMN',
    'SEX',
    'SEX (desc)',
    'BIRTHD',
    'TRIB',
    'TRIB (desc)',
    'OCCU',
    'OCCU (desc)',
    'DLC',
    'STAT',
    'STAT (desc)',
    'MIDN',
    'OBSOLETEFLAGPATIENTTABLE',
    'PATIENTRECORDID',
    'PATIENTUPDATEDBY',
    'PATIENTUPDATEDATE',
    'PATIENTRECORDSTATUS',
    'PATIENTCHECKSTATUS',
    'SURTCT',
    'SURTCT (desc)',
    'SURGERYDATE',
    'CHEMOTREAT',
    'CHEMOTREAT (desc)',
    'CHEMOTREATMENTDATE',
    'IMMUNOTREAT',
    'IMMUNOTREAT (desc)',
    'IMMUNOTREATMENTDATE',
    'HORMONETREATMENT',
    'HORMONETREATMENT (desc)',
    'HORMONETHERAPY',
    'RADIOTREAT',
    'RADIOTREAT (desc)',
    'RADIOTREATMENTDATE',
    'OTHERTREATMENT',
    'OTHERDATE',
    'HOSPITALNUMBER',
    'DEATHCAUSE',
    'REMARKSIFANY',
    'IDENTIFICATIONNO',
    'MARITALSTATUS',
    'MARITALSTATUS (desc)',
    'TELEPHONENO',
    'TELEPHONENUMBERNOK',
    'RELIGION',
    'RELIGION (desc)',
    'EDUCATIONLEVEL',
    'EDUCATIONLEVEL (desc)',
    'SMOKINGSTATUS',
    'SMOKINGSTATUS (desc)',
    'DRINKINGSTATUS',
    'DRINKINGSTATUS (desc)',
  ],
  [
    1,
    'Confirmed',
    1,
    'Done: OK',
    1,
    101,
    'Nairobi',
    'Westlands',
    2018010,
    250,
    'PANCREAS',
    'Head of pancreas',
    9,
    'Unknown',
    8140,
    'Adenocarcinoma, NOS',
    3,
    'Malignant',
    7,
    'Histology of primary',
    null,
    0,
    1,
    'C250',
    '25/11/2019',
    '11f',
    0,
    'KH0150890101',
    'KH015089',
    'KB43743847',
    'carolyne',
    null,
    null,
    9,
    'Unknown',
    null,
    9,
    'Unknown',
    9,
    'Unknown',
    9,
    'Unknown',
    null,
    null,
    9,
    'Unknown',
    9,
    'Unknown',
    'KH015089',
    null,
    'Invalid code.',
    'John',
    'Doe',
    'Jane',
    1,
    'Male',
    '99/99/2016',
    2,
    'Kangemi',
    8,
    'Child',
    '30/11/2018',
    1,
    'Alive',
    'John',
    null,
    'KB43743847',
    'Jean',
    '25/11/2018',
    0,
    null,
    9,
    'Unknown',
    null,
    2,
    'Yes',
    null,
    1,
    'Unknown',
    null,
    9,
    'Unknown',
    null,
    9,
    'Unknown',
    null,
    null,
    null,
    3293232,
    null,
    null,
    null,
    5,
    'Child',
    '7843434343',
    '73223928382 Michael',
    1,
    'Muslim',
    7,
    'Child',
    9,
    'Unknown',
    9,
    'Unknown',
  ],
];

const positions = {
  lefts: [
    'CHEMOTREAT (true)',
    'IMMUNOTREAT (true)',
    'HORMONETREATMENT (true)',
    'RADIOTREAT (true)',
    'SURGERY (true)',
    'HIV(true)',
    'SPECIFICALLYPOSITIVE (true)',
    'OTHERTREATMENT(TRUE)',
    'BAS (true)',
    'TELEPHONENO(+254)',
    'OTHERCONCURRENTILLNESS(true)',
  ],
  rights: [
    'MOR(Matching NCI Codes)',
    'TOP(Matching NCI Codes)',
    'ADDR (Sub County)',
    'WARD',
    'NOKNUMBER',
    'NOKNAME',
    'ADDR (County)',
  ],
  yes: [
    'CHEMOTREAT (true)',
    'IMMUNOTREAT (true)',
    'HORMONETREATMENT (true)',
    'RADIOTREAT (true)',
    'SURGERY (true)',
  ],
};

let visible = true;

const setVisible = () => {
  visible = !visible;
};

const setData = (datas) => {
  data = datas;
};

export { setData, data, visible, setVisible, positions };
