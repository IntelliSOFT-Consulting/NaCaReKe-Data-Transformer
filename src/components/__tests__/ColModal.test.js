import ColModal from '../ColModal';
import codes from '../../NCIcodes';
import { setData, data, visible, setVisible, positions } from '../../tableData';

it('renders correctly', () => {
  const spy = sinon.spy();
  const tree = shallow(
    <ColModal
      handleChange={spy}
      visible={visible}
      setVisible={setVisible}
      cols={data[0]}
      onFinish={spy}
      form={{}}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('shows modal', () => {
  const spy = sinon.spy();
  const tree = shallow(
    <ColModal
      handleChange={spy}
      visible={visible}
      setVisible={setVisible}
      cols={data[0]}
      onFinish={spy}
      form={{}}
    />
  );

  expect(tree.find('Modal').props().visible).toBe(true);
  expect(tree.find('Modal').props().title).toBe('Edit data');
});
