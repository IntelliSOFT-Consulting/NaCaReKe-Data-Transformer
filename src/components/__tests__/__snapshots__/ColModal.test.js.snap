// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  confirmLoading={false}
  okText="Submit"
  okType="primary"
  onCancel={[Function]}
  onOk={[Function]}
  title="Edit data"
  visible={true}
  width={1000}
>
  <ColumnForm
    cols={
      Array [
        "RECS",
        "RECS (desc)",
        "CHEC",
        "CHEC (desc)",
        "AGE",
        "ADDR",
        "ADDR (cat)",
        "ADDR (desc)",
        "INCID",
        "TOP",
        "TOP (cat)",
        "TOP (desc)",
        "LTERALITY",
        "LTERALITY (desc)",
        "MOR",
        "MOR (desc)",
        "BEH",
        "BEH (desc)",
        "BAS",
        "BAS (desc)",
        "MPCODE",
        "MPSEQ",
        "MPTOT",
        "I10",
        "UPDATE",
        "ICCC",
        "OBSOLETEFLAGTUMOURTABLE",
        "TUMOURID",
        "PATIENTIDTUMOURTABLE",
        "PATIENTRECORDIDTUMOURTABLE",
        "TUMOURUPDATEDBY",
        "TUMOURUNDUPLICATIONSTATUS",
        "LABNUMBER",
        "LABREPORTAVAILABLEPTFILE",
        "LABREPORTAVAILABLEPTFILE (desc)",
        "CD4COUNT",
        "STATUSINDICATEDONCLINICALNOTES",
        "STATUSINDICATEDONCLINICALNOTES (desc)",
        "SPECIFICALLYNEGATIVE",
        "SPECIFICALLYNEGATIVE (desc)",
        "SPECIFICALLYPOSITIVE",
        "SPECIFICALLYPOSITIVE (desc)",
        "OTHERCONCURRENTILLNESS",
        "OTHERADDRESS",
        "GRADE",
        "GRADE (desc)",
        "STAGE",
        "STAGE (desc)",
        "REGNO",
        "PERS",
        "PERS (desc)",
        "FIRSTN",
        "MIDDLNAME",
        "FAMN",
        "SEX",
        "SEX (desc)",
        "BIRTHD",
        "TRIB",
        "TRIB (desc)",
        "OCCU",
        "OCCU (desc)",
        "DLC",
        "STAT",
        "STAT (desc)",
        "MIDN",
        "OBSOLETEFLAGPATIENTTABLE",
        "PATIENTRECORDID",
        "PATIENTUPDATEDBY",
        "PATIENTUPDATEDATE",
        "PATIENTRECORDSTATUS",
        "PATIENTCHECKSTATUS",
        "SURTCT",
        "SURTCT (desc)",
        "SURGERYDATE",
        "CHEMOTREAT",
        "CHEMOTREAT (desc)",
        "CHEMOTREATMENTDATE",
        "IMMUNOTREAT",
        "IMMUNOTREAT (desc)",
        "IMMUNOTREATMENTDATE",
        "HORMONETREATMENT",
        "HORMONETREATMENT (desc)",
        "HORMONETHERAPY",
        "RADIOTREAT",
        "RADIOTREAT (desc)",
        "RADIOTREATMENTDATE",
        "OTHERTREATMENT",
        "OTHERDATE",
        "HOSPITALNUMBER",
        "DEATHCAUSE",
        "REMARKSIFANY",
        "IDENTIFICATIONNO",
        "MARITALSTATUS",
        "MARITALSTATUS (desc)",
        "TELEPHONENO",
        "TELEPHONENUMBERNOK",
        "RELIGION",
        "RELIGION (desc)",
        "EDUCATIONLEVEL",
        "EDUCATIONLEVEL (desc)",
        "SMOKINGSTATUS",
        "SMOKINGSTATUS (desc)",
        "DRINKINGSTATUS",
        "DRINKINGSTATUS (desc)",
      ]
    }
    form={Object {}}
    formRef={
      Object {
        "current": null,
      }
    }
    handleChange={[Function]}
    onFinish={[Function]}
  />
</Modal>
`;
